RCTComms/comms.py

```diff
@@ -1022,11 +1078,18 @@ def registerCallback(self, event: EVENTS, callback: Callable) -> None:
                 dictionary shall the the packet payload, and the addr shall be
                 the address of the MAV
         '''
+        callback = self.synchronizedCallback(callback)
         if event.value in self.__packetMap:
             self.__packetMap[event.value].append(callback)
         else:
             self.__packetMap[event.value] = [callback]
 
+    def synchronizedCallback(self, callback):
+        def lockAndCall(packet: rctBinaryPacket, addr: str):
+            with gcsComms.__lock:
+                callback(packet, addr)
+        return lockAndCall
+
```

Comment:
Should this be a wrapper, or should this logic be embedded into the callback logic? i.e. in comms.py:1006
Also, what specifically are we locking access to?

Reply:
Re: locking access
I believe the lock is there to prevent us from somehow sending packets out of order, since we create a new GCSComms object per connection in tower mode and each one spins its own receiver thread. That's why it's a class variable as well (one lock for all the GCSCmms objects). I tried removing all lock functionality to confirm, and the simulator crashed within a few seconds of starting when I ran multiple clients. I can embed the lock into the callback logic though.